rktlet/runtime/pod_sandbox.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	context "golang.org/x/net/context"
+
+	"github.com/golang/glog"
+
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+func formatPod(metaData *runtimeApi.PodSandboxMetadata) string {
+	return fmt.Sprintf("%s_%s(%s)", metaData.Name, metaData.Namespace, metaData.Uid)
+}
+
+func (r *RktRuntime) RunPodSandbox(ctx context.Context, req *runtimeApi.RunPodSandboxRequest) (*runtimeApi.RunPodSandboxResponse, error) {
+	metaData := req.GetConfig().GetMetadata()
+	k8sPodUid := metaData.GetUid()
+	podUUIDFile, err := ioutil.TempFile("", "rktlet_"+k8sPodUid)
+	defer os.Remove(podUUIDFile.Name())
+	if err != nil {
+		return nil, fmt.Errorf("could not create temporary file for rkt UUID: %v", err)
+	}
+
+	// Let the init process to run the pod sandbox.
+	cmd := r.Command("app", "sandbox", "--uuid-file-save="+podUUIDFile.Name())
+	id, err := r.Init.StartProcess(cmd[0], cmd[1:]...)
+	if err != nil {
+		glog.Errorf("failed to run pod %q: %v", formatPod(metaData), err)
+		return nil, err
+
+	}
+
+	glog.V(4).Infof("pod sandbox is running as service %q", id)
+
+	var rktUUID string
+	// TODO, inotify watch for the uuid file would be slightly more efficient
+	// We could also create a pair of pipes for this
+	for i := 0; i < 100; i++ {
+		data, err := ioutil.ReadAll(podUUIDFile)
+		if err != nil {
+			return nil, fmt.Errorf("error reading rkt pod UUID file: %v", err)
+		}
+		if len(data) != 0 {
+			rktUUID = string(data)
+			break
+		}
+
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I think we will need to find better solution for that, but i'm okay for time being - i will create issue about that